index.js

```diff
@@ -314,22 +314,19 @@ module.exports.write = function write(destPath, options) {
         }
       }
 
-      // add new source map file to stream
-      var sourceMapFile = new File({
-        cwd: file.cwd,
-        base: file.base,
-        path: sourceMapPath,
-        contents: new Buffer(JSON.stringify(sourceMap)),
-        stat: {
-          isFile: function () { return true; },
-          isDirectory: function () { return false; },
-          isBlockDevice: function () { return false; },
-          isCharacterDevice: function () { return false; },
-          isSymbolicLink: function () { return false; },
-          isFIFO: function () { return false; },
-          isSocket: function () { return false; }
-        }
-      });
+      // add new source map file to stream, clone from original to keep file.history
```

Comment:
I would recommend adding a new option that flags this as to be done. I do not know all the implications of doing a clone. I would imagine this would add more resource load (this can be heavy already). Therefore I would not make this a default approach and require that this be an option.  Example `sourceMaps.init({doClone:true})` .